### PR TITLE
NMS-13061: Update stat aggregations

### PR DESCRIPTION
**Pull request by cgorantla**
Support timescaledb in Postgres.
Add stat aggregations for BMP

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-13061

### CODE REVIEW

**Comment by opennms-bamboo** [comment]
Welcome CI/CD System (cicd-system@opennms.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username CI/CD System (cicd-system@opennms.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](https://wiki.opennms.org/wiki/images/6/6b/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit https://wiki.opennms.org/wiki/Contributor_Agreement

File: core/schema/src/main/liquibase/28.0.0/changelog.xml [hunk 1]
```diff
@@ -474,7 +477,295 @@
...
+        WHERE relkind='S' AND pg_class.relnamespace = pg_namespace.oid AND relname ILIKE 'bmpipriblognxtid'
+      </sqlCheck>
+    </preConditions>
+    <createSequence sequenceName="bmpipriblognxtid"/>
+    <rollback>
+      <dropSequence sequenceName="bmpipriblognxtid"/>
+    </rollback>
+  </changeSet>
+
+  <changeSet author="cgorantla" id="27.0.0-bmp-ip-rib-log">
```
Comment (christianpape) [inline core/schema/src/main/liquibase/28.0.0/changelog.xml:498]: 27.0.0 for the id? This is the 28.0.0 changelog, so I think it should contain only id starting with 28.

File: features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpMessagePersister.java [hunk 2]
```diff
@@ -268,6 +292,85 @@ void updateGlobalRibsAndAsnInfo() {
...
+            LOG.debug("Stats : Bmp ASN List is empty");
+        } else {
+            LOG.debug("Retrieved {} StatsByAsn elements", statsByAsnList.size());
+        }
+        statsByAsnList.forEach(stat -> {
+            BmpStatsByAsn bmpStatsByAsn = buildBmpStatsByAsn(stat);
+            try {
+                bmpStatsByAsnDao.saveOrUpdate(bmpStatsByAsn);
+            } catch (Exception e) {
+                LOG.error("Exception while persisting BMP Stats by Asn  {}", stat, e);
```
Comment (christianpape) [inline features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpMessagePersister.java:323]: two space before {}

File: features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpMessagePersister.java [hunk 3]
```diff
@@ -268,6 +292,85 @@ void updateGlobalRibsAndAsnInfo() {
...
+            LOG.debug("Stats : Bmp Prefix List is empty");
+        } else {
+            LOG.debug("Retrieved {} StatsByPrefix elements", statsByPrefixList.size());
+        }
+        statsByPrefixList.forEach(stat -> {
+            BmpStatsByPrefix bmpStatsByPrefix = buildBmpStatsByPrefix(stat);
+            try {
+                bmpStatsByPrefixDao.saveOrUpdate(bmpStatsByPrefix);
+            } catch (Exception e) {
+                LOG.error("Exception while persisting BMP Stats by Prefix  {}", stat, e);
```
Comment (christianpape) [inline features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpMessagePersister.java:338]: same here

File: features/telemetry/protocols/bmp/persistence/impl/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/persistence/impl/BmpStatsByPrefixDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import org.opennms.netmgt.telemetry.protocols.bmp.persistence.api.BmpStatsByPrefix;
+import org.opennms.netmgt.telemetry.protocols.bmp.persistence.api.BmpStatsByPrefixDao;
+
+public class BmpStatsByPrefixDaoImpl extends AbstractDaoHibernate<BmpStatsByPrefix, Long> implements BmpStatsByPrefixDao {
+    public BmpStatsByPrefixDaoImpl() {
+        super(BmpStatsByPrefix.class);
+    }
+
+    @Override
+    public BmpStatsByPrefix findByPrefixAndIntervalTime(String peerHashId, String prefix, Integer prefixLen, Date intervalTime) {        Criteria criteria = new Criteria(BmpStatsByPeer.class);
```
Comment (christianpape) [inline features/telemetry/protocols/bmp/persistence/impl/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/persistence/impl/BmpStatsByPrefixDaoImpl.java:47]: Missing line break

**Review by christianpape — CHANGES_REQUESTED** [review]
Only some small things...

**Comment by opennms-bamboo** [comment]
Welcome CI/CD System (cicd-system@opennms.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username CI/CD System (cicd-system@opennms.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](https://wiki.opennms.org/wiki/images/6/6b/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit https://wiki.opennms.org/wiki/Contributor_Agreement

**Comment by cgorantla** [comment]
@christianpape Can you review it one more time. Moved all the scheduled things to Stats module which runs only on OpenNMS and fixed some issues.

**Review by christianpape — APPROVED** [review]
LGTM

**Pull request merged**